src/sql/inspect.py

```diff
@@ -73,6 +74,147 @@ def __init__(self, name, schema, conn=None) -> None:
         self._table_txt = self._table.get_string()
 
 
+@modify_exceptions
+class TableDescription(DatabaseInspection):
+    """
+    Generates descriptive statistics.
+
+    Descriptive statistics are:
+
+    Count - Number of all non empty values
+
+    Mean - Mean of the values
+
+    Max - Maximum of the values in the object.
+
+    Min - Minimum of the values in the object.
+
+    STD - Standard deviation of the observations
+
+    25h, 50h and 75h percentiles
+
+    Unique - Number of unique values
+
+    Top - The most frequent value
+
+    Freq - Frequency of the top value
+
+    """
+
+    def __init__(self, table_name, config=None, user_ns=None) -> None:
+        result_table_columns = sql.run.run(
+            Connection.current, f"SELECT * FROM {table_name} WHERE 1=0", config, user_ns
+        )
+
+        columns = result_table_columns.keys
+
+        table_stats = dict({})
+
+        for column in columns:
+            table_stats[column] = dict()
+            result_col_unique_values = sql.run.run(
+                Connection.current,
+                f"SELECT COUNT(DISTINCT {column}) as unique_count FROM {table_name}",
+                config,
+                user_ns,
+            )
+
+            result_col_freq_values = sql.run.run(
+                Connection.current,
+                f"""SELECT {column}, COUNT({column}) as frequency FROM {table_name}
+                GROUP BY {column} ORDER BY Count({column}) Desc""",
+                config,
+                user_ns,
+            )
+
+            result_non_empty_values = sql.run.run(
+                Connection.current,
+                f"""SELECT {column} FROM {table_name} WHERE {column}
+                IS NOT NULL AND TRIM({column}) <> ''
+                ORDER BY {column} ASC
+                """,
+                config,
+                user_ns,
+            )
+
+            col_values = result_non_empty_values.dict()[column]
+            count = len(col_values)
+            table_stats[column]["count"] = count
+            table_stats[column]["freq"] = result_col_freq_values.dict()["frequency"][0]
+            table_stats[column]["unique"] = result_col_unique_values.dict()[
+                "unique_count"
+            ][0]
+            table_stats[column]["top"] = result_col_freq_values.dict()[column][0]
+            table_stats[column]["min"] = col_values[0]
+            table_stats[column]["max"] = col_values[count - 1]
+
+            try:
+                mean = sum(col_values) / count
```

Comment:
computing the stats this way won't scale, we need tot compute them with SQL directly (we can compute percentiles with `percentile_disc`), check out the plotting module, the code will be similar

Reply:
`percentile_disc` won't work on sqlite. How do we want to handle it?

Currently, if we can't calculate the Nth percentile due to non-numeric value - we show 'nan' in the Nth percentile cell.

I can hide the Nth percentile cell if the database doesn't support the `percentile_disc.

Reply:
yeah, I think it's better to hide it - let's include a message at the end of the table saying that the function isn't supported so the user knows why it doesn't show up